tool/teleport/common/sftp.go

```diff
@@ -515,8 +602,34 @@ func onSFTP() error {
 		return trace.Wrap(err)
 	}
 
+	// Read the file transfer request for this session if one exists
+	bufferedReader := newBufferedReaderCloser(chr)
```

Comment:
Why do we need `newBufferedReaderCloser()` at all? Does this work?

```suggestion
	bufferedReader := bufio.NewReader(chr)
```

The only thing your custom type does that buffered reader doesn't do is close, but we've already deferred a `chr.Close` above.

Reply:
`sftp.NewRequestServer` requires an `io.ReadWriteCloser` as it's first parameter, so passing a `bufio.NewReader` won't work as it doesn't implement `io.Closer` as you already said: https://pkg.go.dev/github.com/pkg/sftp@v1.13.6#NewRequestServer

Reply:
Isn't that what io.NopCloser is for?

Reply:
Will do, missed your last comment before I merged